src/main/scala/loamstream/drm/uger/QacctAccountingClient.scala

```diff
@@ -18,18 +23,57 @@ import scala.sys.process.stringSeqToProcess
  * job id, to facilitate unit testing.
  */
 final class QacctAccountingClient(
+    ugerConfig: UgerConfig,
     val qacctOutputForJobIdFn: String => Seq[String]) extends AccountingClient with Loggable {
 
   import QacctAccountingClient._
 
   //Memoize the function that retrieves the metadata, to avoid running something expensive, like invoking
   //qacct in the production case, more than necessary.
-  //NB: Don't cache empty results, since this likely indicates a failure when invoking qaact, and we'd like to
-  //be able to try again in that case.
+  //NB: If qacct fails, retry up to ugerConfig.maxQacctRetries times, waiting 0.5, 1, 2, 4, ... up to 30s 
+  //in between each one.
   private val qacctOutputForJobId: String => Seq[String] = {
-    val shouldCache: Seq[String] => Boolean = _.nonEmpty
+    val doRetries: String => Seq[String] = { jobId =>
+      val maxRuns = ugerConfig.maxQacctRetries + 1
+      
+      def invokeQacct(): Try[Seq[String]] = Try(qacctOutputForJobIdFn(jobId))
+      
+      val delays = delaySequence
+      
+      def delayIfFailure(attempt: Try[Seq[String]]): Try[Seq[String]] = {
+        if(attempt.isFailure) {
+          //TODO: evaluate whether or not blocking is ok. For now, it's expedient and doesn't seem to cause problems.
+          Thread.sleep(delays.next().toMillis)
+        }
+        
+        attempt
+      }
+      
+      val attempts = Iterator.continually(invokeQacct()).take(maxRuns).map(delayIfFailure).dropWhile(_.isFailure)
```

Comment:
Okay... this is slick. Adding to my mental "bag of things to remember..."

Reply:
Thanks.  `Iterator` has taken over 80-90% of the remaining times I'd want to use a `while` and some `var`s.